helpers/setupjsongenerator.py

```
@@ -0,0 +1,675 @@
+        """
+        Remove given disk from the specified storagerouter
+        :param storagerouter_ip: storagerouter to remove disk from
+        :type storagerouter_ip: str
+        :param name: name of the disk to be removed
+        :type name: str
+        """
+        try:
+            self.config['storagerouters'][storagerouter_ip]['disks'].pop(name)
+        except Exception:
+            pass
+
+    def add_domain_to_sr(self, storagerouter_ip, name, recovery=False):
+        """
+        Add domains, present in the model, to a storage router.
+        :param storagerouter_ip: ip of the storage router
+        :type storagerouter_ip: str
+        :param name: name of the domain to add to the storagerouter
+        :type name: str
+        :param recovery: true or false whether the domain is a recovery domain or not
+        :type recovery: bool
+        """
+        self._valid_storagerouter(storagerouter_ip)
+        Toolbox.verify_required_params(required_params={'name': (str, None, True)}, actual_params={'name': name}, verify_keys=True)
+
+        if name not in self._domains:
+            raise ValueError('Invalid domain passed: {0}'.format(name))
+
+        path = self.config['storagerouters'][storagerouter_ip]
+        if 'domains' not in path.keys():
+            path['domains'] = {}
+        path = path['domains']
+        config_key = 'domain_guids' if recovery is False else 'recovery_domain_guids'
+        if config_key not in path:
+            path[config_key] = []
+        path[config_key].append(name)
+
+    def remove_domain_from_sr(self, storagerouter_ip, name):
+        """
+        Remove the given domain from the storagerouter
+        :param storagerouter_ip: storagerouter to remove the domains from
+        :type storagerouter_ip: str
+        :param name: name of the domain to remove
+        :type name: str
+        """
+        try:
+            _ = self.config['storagerouters'][storagerouter_ip]['domains']['domain_guids']
```

why not add the .remove to the end of this chain? You hide any exception anyway